job/server/src/main/java/alluxio/master/job/workflow/WorkflowTracker.java

```diff
@@ -250,6 +250,7 @@ private synchronized void next(long jobId) {
         mJobMaster.run(childJobConfig, childJobId);
       } catch (JobDoesNotExistException | ResourceExhaustedException e) {
         LOG.warn(e.getMessage());
+        workflowExecution.stop(Status.FAILED, e.getMessage());
```

Comment:
Should this logic be in the `stop` call? Does the bug happen for workflows which fail through `PlanTracker#onPlanStatusChange`?

Reply:
No the bug doesn't happen for workflows which fail through PlanTracker#onPlanStatusChange because all logic that goes through onPlanStatusChange are not workflows and are Plans which are handled by the PlanTracker. This is also the reason why the stop logic looks at the parent. onPlanStatusChange is the normal case and this JobDoesNotExist + ResourceExhausted is the edge case.